router/router.go

```diff
@@ -977,17 +997,40 @@ func (worker *workerT) updateReqMetrics(respStatusCode int, diagnosisStartTime *
 	worker.rt.trackRequestMetrics(reqMetric)
 }
 
-func (worker *workerT) updateAbortedMetrics(destinationID, statusCode string) {
+type allowedRtTags struct {
+	allowTf  string
+	allowDel string
+}
+
+func (worker *workerT) getAllowedRtTags(errorAt string) allowedRtTags {
+	// when error occur during transformation(rt or batch)
+	// when proxy is not enabled or when rtAbortDeliveryAlert is true(default)
+	allowTf := errorAt == ERROR_AT_TF && worker.rt.allowRtAbortForTf
+	// (when error occur during delivery
+	// when proxy is not enabled or when rtAbortDeliveryAlert is true(default))
+	// or when destination is managed custom destination manager
+	allowDel := (errorAt == ERROR_AT_DEL && (!worker.rt.transformerProxy || worker.rt.allowRtAbortAlertForDelv)) || errorAt == ERROR_AT_CUST
+	return allowedRtTags{
+		allowTf:  strconv.FormatBool(allowTf),
+		allowDel: strconv.FormatBool(allowDel),
+	}
+}
+
+func (worker *workerT) updateAbortedMetrics(destinationID, statusCode, errorAt string) {
+	allowedTags := worker.getAllowedRtTags(errorAt)
 	eventsAbortedStat := stats.Default.NewTaggedStat(`router_aborted_events`, stats.CountType, stats.Tags{
 		"destType":       worker.rt.destName,
 		"respStatusCode": statusCode,
 		"destId":         destinationID,
+		"allowTf":        allowedTags.allowTf,
+		"allowDel":       allowedTags.allowDel,
```

Comment:
Why do we need 2 tags for alerting, wouldn't a single `alert` tag suffice? I also find the choice of names a bit unintuitive

Reply:
We already have 2 configuration options in the server for controlling when an alert should be sent, why do we need two flags for the alert definition itself too?

Another point is that the purpose of an `alert` tag would be pretty obvious to anyone, whereas I still find it hard to be able to easily decode the meaning of `allowTf` & `allowDel` tags.

Reply:
>why do we need two flags for the alert definition itself too?

Basically we want the old alert definition to co-exist with the new one. In that case, we don't want duplicate alerts to be triggered. Hence by sending these tags to the measurement, we are making sure that only alerts that cannot be alerted from new definition should be alerted from older one.

These tags will be used for older alert definition(in operator) so that only those alerts that need to be sent via the older one would be used. Let me know if it clarifies the question

> allowTf & allowDel tags

Does `alertOnTransformation` and `alertOnDelivery` communicate the purpose & is easily understood ?